conda_lock/conda_lock.py

```diff
@@ -542,7 +542,7 @@ def render_lockfile_for_platform(  # noqa: C901
     lockfile.toposort_inplace()
 
     for p in lockfile.package:
-        if p.platform == platform and ((not p.optional) or (p.category in categories)):
+        if p.platform == platform and p.category in categories:
```

Comment:
Note to self: `not p.optional` ⇒ `p.category == "main"`.
Since `"main" in categories`, this is redundant and can be deleted.

Therefore, this is logically sound.